packages/file/lib/src/backends/local/local_file_system.dart

```diff
@@ -31,7 +35,9 @@ class LocalFileSystem extends FileSystem {
   Link link(dynamic path) => LocalLink(this, io.Link(getPath(path)));
 
   @override
-  p.Context get path => p.Context();
+  p.Context get path {
+    return _expandoContext[this] ??= p.Context();
```

Comment:
...so instead of using an expando here, we return a _LocalFileSystemContext(this), which manages the creation of a context and only creates teh context anew for things that might change like cwd.

Reply:
And AFAICT `.current` is the only thing that should end up triggering a re-create.